src/system/SystemLayerImplFreeRTOS.cpp

```diff
@@ -76,6 +76,24 @@ CHIP_ERROR LayerImplFreeRTOS::StartTimer(Clock::Timeout delay, TimerCompleteCall
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR LayerImplFreeRTOS::ExtendTimerTo(Clock::Timeout delay, TimerCompleteCallback onComplete, void * appState)
+{
+    assertChipStackLockedByCurrentThread();
+    Clock::Timestamp remainingTime = mTimerList.GetRemainingTime(onComplete, appState);
+
+    if (remainingTime.count() < Clock::Timestamp(delay).count())
```

Comment:
This doesn't make sense.  On the left we have a _timestamp_.  On the right we have a _duration_.

Was this tested?

Reply:
The return type.of remainingTime is wrong. It the timestamp - the current timestamps. This code should be ok. I'll fix the type